lib/flutter/flutter_daemon.dart

```diff
@@ -51,13 +63,11 @@ class FlutterDaemonManager implements Disposable {
         .where((String str) => str.startsWith('[') && str.endsWith(']'))
         .map((String str) => str.substring(1, str.length - 1));
 
-      // process.onStderr.listen((String str) {
-      //   _logger.fine('stderr: $str');
-      // });
-
-      // TODO: Do we want to dispose on exit? Notify someone?
       process.onExit.then((_) {
         _daemon.dispose();
```

Comment:
It looks like there was an issue if we received two non-null sdk events in a row. We'd initialize the daemon the first time, and then initialize a 2nd one for the 2nd event, leaving the first one orphaned. I've added a guard check to make sure we only spin up a daemon if there is no current one.